src/SerilogWeb.Classic/Classic/SerilogWebClassicConfiguration.cs

```diff
@@ -89,7 +89,7 @@ internal SerilogWebClassicConfiguration Edit(Func<SerilogWebClassicConfiguration
         /// <param name="value">Value of the pair</param>
         internal string FilterPasswords(string key, string value)
         {
-            if (FilterPasswordsInFormData && FilteredKeywordsInFormData.Any(keyword => key.IndexOf(keyword, StringComparison.OrdinalIgnoreCase) != -1))
+            if (FilterPasswordsInFormData && key != null && FilteredKeywordsInFormData.Any(keyword => key.IndexOf(keyword, StringComparison.OrdinalIgnoreCase) != -1))
```

Comment:
this `boolean` condition is starting to get a bit out of hand, with a combination of positive and negative conditions, it's starting to be a bit long and hard to grasp. 

Maybe we could refactor it and have it in multiple lines with *early return* instead ? 

Something like : 

```cs
if(!FilterPasswordsInFormData){
  return value;
}
if(key == null){
  return value;
}
if(FilteredKeywordsInFormData.Any(keyword => key.IndexOf(keyword, StringComparison.OrdinalIgnoreCase) != -1)){
  return "********";
}
return value;
```

or maybe another approach would be extracting the complicated `key.IndexOf(keyword, StringComparison.OrdinalIgnoreCase) != -1)` to its own method like `private static boolean ContainsIgnoringCase(string text, string substring)` , and let it just return `false` when `substring` is `null`. 

What do you think ?

Reply:
Should we just split the if statements? It feels like that will negate the need for extracting the IndexOf into a separate method. In a way, adding another method introduces another type of complexity by making a developer dig further into code to understand what it's doing.